test/metabase/test/data/mbql_query_impl.clj

```diff
@@ -8,7 +8,18 @@
 
 (set! *warn-on-reflection* true)
 
-(def ^:dynamic *id-fn-symb*              'metabase.test.data/id)
+#_:clj-kondo/ignore
+(defn druid-id-fn
+  "I have a strong feeling this should be handled differently! Let's discuss that during review!"
+  [& args]
+  (if (and (> (count args) 1)
+           (= :druid-jdbc @(requiring-resolve 'metabase.driver/*driver*))
+           (= "timestamp" (name (last args))))
+    (apply @(requiring-resolve 'metabase.test.data/id) (conj (vec (butlast args)) :__time))
```

Comment:
```suggestion
    (apply (requiring-resolve 'metabase.test.data/id) (conj (vec (butlast args)) :__time))
```

`requiring-resolve` returns a `clojure.lang.Var` and you can invoke a var just like a regular function, you don't need to `deref` it first.